src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextInputFormatter.cs

```diff
@@ -15,6 +16,53 @@ namespace Microsoft.AspNetCore.Mvc.Formatters
     /// </summary>
     public abstract class TextInputFormatter : InputFormatter
     {
+        // ASP.NET Core MVC 1.0 used Encoding.GetEncoding() when comparing the request charset with SupportedEncodings.
+        // That method supports the following alternate names for system-supported encodings. This table maps from
+        // .NET Core's keys in s_encodingDataTable to values in s_encodingDataTableItems, less identity mappings. It
+        // should be kept in sync with those Unix-specific tables in
+        // https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs
+        // or their Windows equivalents (also for .NET Core not desktop .NET): COMNlsInfo::EncodingDataTable and
+        // COMNlsInfo::CodePageDataTable in EncodingData in
+        // https://github.com/dotnet/coreclr/blob/master/src/classlibnative/nls/encodingdata.cpp
+        private static readonly IReadOnlyDictionary<string, string> EncodingAliases = new Dictionary<string, string>
+        {
+            { "ANSI_X3.4-1968", "us-ascii" },
+            { "ANSI_X3.4-1986", "us-ascii" },
+            { "ascii", "us-ascii" },
+            { "cp367", "us-ascii" },
+            { "cp819", "iso-8859-1" },
+            { "csASCII", "us-ascii" },
+            { "csISOLatin1", "iso-8859-1" },
+            { "csUnicode11UTF7", "utf-7" },
+            { "IBM367", "us-ascii" },
+            { "ibm819", "iso-8859-1" },
+            { "ISO-10646-UCS-2", "utf-16" },
+            { "iso-ir-100", "iso-8859-1" },
+            { "iso-ir-6", "us-ascii" },
+            { "ISO646-US", "us-ascii" },
+            { "ISO_646.irv:1991", "us-ascii" },
+            { "iso_8859-1:1987", "iso-8859-1" },
+            { "l1", "iso-8859-1" },
+            { "latin1", "iso-8859-1" },
+            { "ucs-2", "utf-16" },
+            { "unicode", "utf-16"},
+            { "unicode-1-1-utf-7", "utf-7" },
+            { "unicode-1-1-utf-8", "utf-8" },
+            { "unicode-2-0-utf-7", "utf-7" },
+            { "unicode-2-0-utf-8", "utf-8" },
+            // People get confused about the FFFE here.  We can't change this because it'd break existing apps.
+            // This has been this way for a long time, including in Mlang.
+            // Big Endian, BOM seems backwards, think of the BOM in little endian order.
```

Comment:
Copied this comment from https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs but lean toward removing it here. Doesn't seem that useful. Thoughts?